File: src/NoiseSentinel.DAL/Models/Court.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace NoiseSentinel.DAL.Models;

[Table("COURT")]
public partial class Court
{
    [Key]
    [Column("CourtID")]
    public int CourtId { get; set; }

    [StringLength(255)]
    public string? CourtName { get; set; }

    [Column("CourtTypeID")]
    public int? CourtTypeId { get; set; }

    [StringLength(255)]
    public string? Location { get; set; }

    [StringLength(255)]
    public string? District { get; set; }

    [StringLength(255)]
    public string? Province { get; set; }

    [ForeignKey("CourtTypeId")]
    [InverseProperty("Courts")]
    public virtual Courttype? CourtType { get; set; }

    [InverseProperty("Court")]
    public virtual ICollection<Judge> Judges { get; set; } = new List<Judge>();
}

File: src/Noisesentinel.WebPortal/src/models/Court.ts
/**
 * Court-related TypeScript models/interfaces
 * Based on NoiseSentinel.BLL.DTOs.Court namespace
 */

// Court Type
export interface CourtType {
  courtTypeId: number;
  courtTypeName: string;
}

// Judge Summary (for court response)
export interface JudgeSummary {
  judgeId: number;
  fullName: string;
  rank: string | null;
  serviceStatus: boolean;
}

// Court List Item
export interface CourtListItem {
  courtId: number;
  courtName: string;
  courtTypeName: string;
  location: string;
  district: string;
  province: string;
  totalJudges: number;
}

// Court Response (Full Details)
export interface CourtResponse {
  courtId: number;
  courtName: string;
  courtTypeId: number;
  courtTypeName: string;
  location: string;
  district: string;
  province: string;
  totalJudges: number;
  judges: JudgeSummary[];
}

// Create Court DTO
export interface CreateCourtDto {
  courtName: string;
  courtTypeId: number;
  location: string;
  district: string;
  province: string;
}

// Update Court DTO
export interface UpdateCourtDto {
  courtId: number;
  courtName: string;
  courtTypeId: number;
  location: string;
  district: string;
  province: string;
}

// Court Statistics
export interface CourtStatistics {
  totalCourts: number;
  supremeCourts: number;
  highCourts: number;
  districtCourts: number;
  civilCourts: number;
  sessionsCourts: number;
  totalJudges: number;
  totalCases: number;
}
